server/src/main/java/org/hl7/davinci/endpoint/JWTAuthorizationFilter.java

```
@@ -0,0 +1,120 @@
+    return null;
+  }
+
+  /**
+   * Checks if the token is well formed.  This includes following
+   * rfc7515 standard for JWT and the CDS-Hooks specification.
+   *
+   * @param token the JWT to be checked
+   * @return a boolean indicating whether the token is well formed
+   */
+  private String isTokenWellFormed(String token) {
+    JsonParser parser = new JsonParser();
+    // JWT tokens are of the form:
+    //    header     payload     signature
+    //   xxxxxxxxx.yyyyyyyyyyy.zzzzzzzzzzzz
+    // where each section is base64 encoded JSON.
+    // We need to check if the header and the payload
+    // have the correct keys and if the signature is present.
+    // The signature will be validated later.
+
+    String[] tokens = token.split("\\.");
+    if (tokens.length != 3) {
+      // tokens without all three sections are invalid
+      return null;
+    }
+    try {
+      String header = new String(Base64.getDecoder().decode(tokens[0]));
+      String payload = new String(Base64.getDecoder().decode(tokens[1]));
+
+      // we just want to make sure the json has the correct fields
+      // their contents aren't really important, if the contents are
+      // bad we can reject them later.
+      List<String> requiredHeaders = Arrays.asList("alg","typ","kid");
+      List<String> requiredPayload = Arrays.asList("iss","aud","exp","iat","jti");
+
+      JsonObject jsonHeader = parser
+          .parse(header)
+          .getAsJsonObject();
+      boolean validHeader = jsonHeader
+          .keySet()
+          .containsAll(requiredHeaders);
+
+      JsonObject jsonPayload = parser
+          .parse(payload)
+          .getAsJsonObject();
+      boolean validPayload = jsonPayload
+          .keySet()
+          .containsAll(requiredPayload);
+
+      if (validHeader && validPayload) {
+        return jsonPayload.get("iss").toString();
+      } else {
+        return null;
+      }
+
+    } catch (Exception e) {
```

Can you make this less broad than catching `Exception`? If anything goes wrong in this method, it silently fails and returns `null`. This could make troubleshooting really hard later.